server/api/dto/request/requests.go

```diff
@@ -21,9 +28,65 @@ type UpdateCredentialsDto struct {
 	Name         string `json:"name" validate:"required"`
 }
 
+type WebauthnRequests interface {
+	InitRegistrationDto | InitTransactionDto
+}
+
 type InitRegistrationDto struct {
 	UserId      string  `json:"user_id" validate:"required"`
 	Username    string  `json:"username" validate:"required,max=128"`
 	DisplayName *string `json:"display_name,max=128"`
 	Icon        *string `json:"icon"`
 }
+
+func (initRegistration *InitRegistrationDto) ToModel() *models.WebauthnUser {
+	icon := ""
+	if initRegistration.Icon != nil {
+		icon = *initRegistration.Icon
+	}
+
+	displayName := initRegistration.Username
+	if initRegistration.DisplayName != nil {
+		displayName = *initRegistration.DisplayName
+	}
+
+	webauthnId, _ := uuid.NewV4()
+
+	now := time.Now()
+
+	return &models.WebauthnUser{
+		ID:          webauthnId,
+		UserID:      initRegistration.UserId,
+		Name:        initRegistration.Username,
+		Icon:        icon,
+		DisplayName: displayName,
+		CreatedAt:   now,
+		UpdatedAt:   now,
+	}
+}
+
+type InitTransactionDto struct {
+	UserId          string      `json:"user_id" validate:"required"`
+	TransactionId   string      `json:"transaction_id" validate:"required"`
```

Comment:
In the open-api spec it states, that the `transaction_id` must not be longer than 128 characters, this should/can be checked with a validation tag.

Reply:
Fixed